src/patterns/gen-ai/aws-text-to-sql/index.ts

```diff
@@ -0,0 +1,1219 @@
+    } else if (props?.eventBusProps) {
+      this.eventBus = eventBridgeHelper.buildEventBus(this, {
+        eventBusProps: props.eventBusProps,
+      });
+    } else {
+      this.eventBus = new events.EventBus(
+        this,
+        'textToSqlBus' + props.stage,
+        {
+          eventBusName: generatePhysicalNameV2(
+            this,
+            'textToSql' + this.stage,
+            { maxLength: 63, lower: true },
+          ),
+        },
+      );
+    }
+
+    this.stepFunction = new stepfunctions.StateMachine(
+      this,
+      'textToSqlStepFunction' + props.stage,
+      {
+        definition: reformulateQuestionState.next(
+          feedbackChoiceStateOne
+            .when(
+              stepfunctions.Condition.stringEquals(
+                '$.semantic_layer_strategy',
+                'human',
+              ),
+              reformulatedQuestionFeedbackState.next(queryGeneratorState),
+            )
+            .otherwise(queryGeneratorState),
+        ),
+        timeout: Duration.days(90),
+      },
+    );
+
+
+    this.eventsRule = new events.Rule(this, 'EventsRule', {
+      eventBus: this.eventBus,
+      targets: [new eventsTarget.SfnStateMachine(
+        this.stepFunction, {
+          role: eventsRole,
+        },
+      )],
+      eventPattern: {
+        source: ['webclient'],
+      },
+    });
+
+    this.eventBus.grantPutEventsTo(this.stepFunction);
+
+    // Grant the start execution permission to the Events service
+    this.stepFunction.grantStartExecution(eventsRole);
+  } // end construct
+
+  private validateDbProps(props: TextToSqlProps): void {
+    // Check if existingAuroraDbCluster and databaseClusterProps are set at the same time
+    if (props.existingAuroraDbCluster && props.databaseClusterProps) {
+      throw new Error(
+        'Only one of existingAuroraDbCluster or databaseClusterProps can be set at a time.',
+      );
+    }
+
+    // Check if databaseInstanceProps and existingRdsDbInstance are set at the same time
+    if (props.databaseInstanceProps && props.existingRdsDbInstance) {
+      throw new Error(
+        'Only one of databaseInstanceProps or existingRdsDbInstance can be set at a time.',
+      );
+    }
+
+    // Check if existingAuroraDbCluster and existingRdsDbInstance are set at the same time
+    if (props.existingAuroraDbCluster && props.existingRdsDbInstance) {
+      throw new Error(
+        'Only one of existingAuroraDbCluster or existingRdsDbInstance can be set at a time.',
+      );
+    }
+  }
+
+  private createSecret(): secretsmanager.Secret {
+    return new secretsmanager.Secret(this, 'AuroraSecret' + this.stage, {
+      secretName: 'texttosqldbsecret',
+      generateSecretString: {
+        excludePunctuation: true,
+        passwordLength: 16,
+        secretStringTemplate: JSON.stringify({
+          username: 'admin',
+        }),
+        generateStringKey: 'password',
+      },
+    });
+  }
+
+  private createRdsInstance(
```

Comment:
For both methods below, I'm surprised there is no cdk nag errors triggered when tests are running. See https://github.com/cdklabs/cdk-nag/blob/main/RULES.md for AwsSolutions-RDS* rules. 
I think we either:
- make the construct non contained, a.k.a we never create the DBs on behalf of the user, and the user needs to provide an existing instance to the construct. We can show an example of DB provisioning in the sample app
- if we allow the construct to create those resources on behalf of the user, we need to make sure they follow the best practices, and mention the whole configuration in the README file

Reply:
option 1 make sense. Did the changes - d28bbe26726f810317c2b3533f33c248fd7c9cfb